src/NHibernate/Logging.cs

```diff
@@ -222,16 +223,16 @@ public void WarnFormat(string format, params object[] args)
 	public class Log4NetLoggerFactory: ILoggerFactory
 	{
 		private static readonly System.Type LogManagerType = System.Type.GetType("log4net.LogManager, log4net");
-		private static readonly Func<string, object> GetLoggerByNameDelegate;
+		private static readonly Func<Assembly, string, object> GetLoggerByNameDelegate;
 		private static readonly Func<System.Type, object> GetLoggerByTypeDelegate;
 		static Log4NetLoggerFactory()
 		{
-			GetLoggerByNameDelegate = GetGetLoggerMethodCall<string>();
+			GetLoggerByNameDelegate = GetGetLoggerByNameMethodCall();
 			GetLoggerByTypeDelegate = GetGetLoggerMethodCall<System.Type>();
 		}
 		public IInternalLogger LoggerFor(string keyName)
 		{
-			return new Log4NetLogger(GetLoggerByNameDelegate(keyName));
+			return new Log4NetLogger(GetLoggerByNameDelegate(typeof(Log4NetLoggerFactory).Assembly, keyName));
```

Comment:
Don't we need an overload for specifying another `Assembly` than NHibernate.dll? They are two calls in NHibernate.Test.dll which were likely resolving to the test assembly instead.